govcd/nsxt_distributed_firewall.go

```diff
@@ -0,0 +1,88 @@
+package govcd
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/vmware/go-vcloud-director/v2/types/v56"
+)
+
+// DistributedFirewall contains a types.DistributedFirewallRules which handles Distributed Firewall
+// rules in a VDC Group
+type DistributedFirewall struct {
+	DistributedFirewallRuleContainer *types.DistributedFirewallRules
+	client                           *Client
+	VdcGroup                         *VdcGroup
+}
+
+// GetDistributedFirewall retrieves Distributed Firewall in a VDC Group which contains all rules
+func (vdcGroup *VdcGroup) GetDistributedFirewall() (*DistributedFirewall, error) {
+	client := vdcGroup.client
+	endpoint := types.OpenApiPathVersion1_0_0 + types.OpenApiEndpointVdcGroupsDfwRules
+	apiVersion, err := client.getOpenApiHighestElevatedVersion(endpoint)
+	if err != nil {
+		return nil, err
+	}
+
+	urlRef, err := client.OpenApiBuildEndpoint(fmt.Sprintf(endpoint, vdcGroup.VdcGroup.Id, "default"))
+	if err != nil {
+		return nil, err
+	}
+
+	returnObject := &DistributedFirewall{
+		DistributedFirewallRuleContainer: &types.DistributedFirewallRules{},
+		client:                           client,
+		VdcGroup:                         vdcGroup,
+	}
+
+	err = client.OpenApiGetItem(apiVersion, urlRef, nil, returnObject.DistributedFirewallRuleContainer, nil)
+	if err != nil {
+		return nil, fmt.Errorf("error retrieving Distributed Firewall rules: %s", err)
+	}
+
+	return returnObject, nil
+}
+
+// UpdateDistributedFirewall updates Distributed Firewall in a VDC Group
+func (vdcGroup *VdcGroup) UpdateDistributedFirewall(dfwRules *types.DistributedFirewallRules) (*DistributedFirewall, error) {
+	client := vdcGroup.client
+	endpoint := types.OpenApiPathVersion1_0_0 + types.OpenApiEndpointVdcGroupsDfwRules
+	apiVersion, err := client.getOpenApiHighestElevatedVersion(endpoint)
+	if err != nil {
+		return nil, err
+	}
+
+	// "default" policy is hardcoded because there is no other policy supported
```

Comment:
Maybe we could put this comment also in line 27?

Reply:
Added this and `note` for each function docs.